src/openrct2/localisation/Localisation.h

```diff
@@ -98,13 +98,41 @@ template<typename T> void constexpr set_format_arg(uint8_t* args, size_t offset,
         set_format_arg_body(gCommonFormatArgs, offset, (uintptr_t)(value), sizeof(type));                                      \
     } while (false)
 
-#define set_format_arg_on(args, offset, type, value) set_format_arg_body(args, offset, (uintptr_t)(value), sizeof(type))
-
 #define set_map_tooltip_format_arg(offset, type, value)                                                                        \
     do                                                                                                                         \
     {                                                                                                                          \
         static_assert(sizeof(type) <= sizeof(uintptr_t), "Type too large");                                                    \
         set_format_arg_body(gMapTooltipFormatArgs, offset, (uintptr_t)(value), sizeof(type));                                  \
     } while (false)
 
+class Formatter
+{
+    const uint8_t* _start;
+    uint8_t* _end;
+
+public:
+    Formatter(uint8_t* buf)
+        : _start(buf)
+        , _end(buf)
+    {
+    }
+
+    auto buf()
+    {
+        return _end;
+    }
+
+    std::size_t bytes() const
+    {
+        return _end - _start;
+    }
+
+    template<typename T, typename V> Formatter& add(V value)
```

Comment:
Following the short discussion on #11354 , these should probably receive more meaningful names and be preceded with `T`